core/src/main/scala/org/apache/spark/deploy/history/EventLogFileCompactor.scala

```diff
@@ -0,0 +1,235 @@
+    val results = filters.flatMap(filter => applyFilter(filter, event))
+    results.isEmpty || results.forall(_ == true)
+  }
+
+  private def applyFilter(filter: EventFilter, event: SparkListenerEvent): Option[Boolean] = {
+    event match {
+      case event: SparkListenerStageSubmitted => filter.filterStageSubmitted(event)
+      case event: SparkListenerStageCompleted => filter.filterStageCompleted(event)
+      case event: SparkListenerJobStart => filter.filterJobStart(event)
+      case event: SparkListenerJobEnd => filter.filterJobEnd(event)
+      case event: SparkListenerTaskStart => filter.filterTaskStart(event)
+      case event: SparkListenerTaskGettingResult => filter.filterTaskGettingResult(event)
+      case event: SparkListenerTaskEnd => filter.filterTaskEnd(event)
+      case event: SparkListenerEnvironmentUpdate => filter.filterEnvironmentUpdate(event)
+      case event: SparkListenerBlockManagerAdded => filter.filterBlockManagerAdded(event)
+      case event: SparkListenerBlockManagerRemoved => filter.filterBlockManagerRemoved(event)
+      case event: SparkListenerUnpersistRDD => filter.filterUnpersistRDD(event)
+      case event: SparkListenerApplicationStart => filter.filterApplicationStart(event)
+      case event: SparkListenerApplicationEnd => filter.filterApplicationEnd(event)
+      case event: SparkListenerExecutorMetricsUpdate => filter.filterExecutorMetricsUpdate(event)
+      case event: SparkListenerStageExecutorMetrics => filter.filterStageExecutorMetrics(event)
+      case event: SparkListenerExecutorAdded => filter.filterExecutorAdded(event)
+      case event: SparkListenerExecutorRemoved => filter.filterExecutorRemoved(event)
+      case event: SparkListenerExecutorBlacklistedForStage =>
+        filter.filterExecutorBlacklistedForStage(event)
+      case event: SparkListenerNodeBlacklistedForStage =>
+        filter.filterNodeBlacklistedForStage(event)
+      case event: SparkListenerExecutorBlacklisted => filter.filterExecutorBlacklisted(event)
+      case event: SparkListenerExecutorUnblacklisted => filter.filterExecutorUnblacklisted(event)
+      case event: SparkListenerNodeBlacklisted => filter.filterNodeBlacklisted(event)
+      case event: SparkListenerNodeUnblacklisted => filter.filterNodeUnblacklisted(event)
+      case event: SparkListenerBlockUpdated => filter.filterBlockUpdated(event)
+      case event: SparkListenerSpeculativeTaskSubmitted =>
+        filter.filterSpeculativeTaskSubmitted(event)
+      case _ => filter.filterOtherEvent(event)
+    }
+  }
+}
+
+class CompactedEventLogFileWriter(
+    originalFilePath: Path,
+    appId: String,
+    appAttemptId: Option[String],
+    logBaseDir: URI,
+    sparkConf: SparkConf,
+    hadoopConf: Configuration)
+  extends SingleEventLogFileWriter(appId, appAttemptId, logBaseDir, sparkConf, hadoopConf) {
+
+  override val logPath: String = originalFilePath.toUri.toString + EventLogFileWriter.COMPACTED
```

Comment:
Maybe we can call such vals `COMPACTED_SUFFIX`? When I see `stripSuffix` I know what it is but such case when stays alone maybe better to name it more meaningful.